### PR TITLE
[oneTBB] Improve oneTBB type & named requirements

### PR DESCRIPTION
**Pull request by aepanchi**

### CODE REVIEW

**Comment by rscohn2** [comment]
Is this ready for merging?

File: source/elements/oneTBB/source/algorithms/functions/parallel_for_each_func.rst [hunk 1]
```diff
@@ -37,8 +37,11 @@ Function template that processes work items in parallel.
 Requirements:
 
 * The ``Body`` type must meet the :doc:`ParallelForEachBody requirements <../../named_requirements/algorithms/par_for_each_body>`.
-* The ``InputIterator`` type must meet the `Input Iterator` requirements from the [input.iterators] ISO C++ Standard section.
+* The ``InputIterator`` type must meet the `Input Iterator` requirements from the [input.iterators] section of the ISO C++ Standard.
 * The ``Container`` type must meet the :doc:`ContainerBasedSequence requirements <../../named_requirements/algorithms/container_based_sequence>`.
+* If ``InputIterator`` type does not meet the `Forward Iterator` requirements from the [forward.iterators] section of the ISO C++ Standard,
+  the ``std::iterator_traits<InputIterator>::value_type`` type must be constructible from ``std::iterator_traits<InputIterator>::reference``.
```
Comment (akukanov) [inline source/elements/oneTBB/source/algorithms/functions/parallel_for_each_func.rst:43]: Move these lines above, right after the line 40

File: source/elements/oneTBB/source/algorithms/functions/parallel_for_each_func.rst [hunk 2]
```diff
@@ -37,8 +37,11 @@ Function template that processes work items in parallel.
...
+* ``decltype(std::declval<Container>().begin())`` must meet the same requirements as the `Input Iterator` type.
```
Comment (akukanov) [inline source/elements/oneTBB/source/algorithms/functions/parallel_for_each_func.rst:44]:
```suggestion
* The type returned by ``Container::begin()`` must meet the same requirements as the ``InputIterator`` type above.
```

Comment (kboyarinov) [reply source/elements/oneTBB/source/algorithms/functions/parallel_for_each_func.rst]: Done

File: source/elements/oneTBB/source/algorithms/functions/parallel_for_each_func.rst (hunk 1 above)
Comment (kboyarinov) [reply source/elements/oneTBB/source/algorithms/functions/parallel_for_each_func.rst:42]: Done

File: source/elements/oneTBB/source/named_requirements/algorithms/par_for_index.rst [hunk 3]
```diff
@@ -25,31 +25,48 @@ A type `Index` satisfies `ParallelForIndex` if it meets the following requiremen
...
+
+.. cpp:function:: Index operator/( const Index& i, const Index& j )
+
+    Quotient of *i* and *j* values.
+
 ``D`` is the type of the expression ``j-i``. It can be any integral type that is convertible to ``size_t``.
 Examples that model the Index requirements are integral types and pointers.
 
+**_NOTE:_** It is recommended to use integral types as ``ParallelForIndex``. See ``[basic.fundamental]``
+ISO C++ Standard section for information about integral types.
```
Comment (akukanov) [inline source/elements/oneTBB/source/named_requirements/algorithms/par_for_index.rst:68]:
```suggestion
**_NOTE:_** It is recommended to use integral types as ``ParallelForIndex``. See the ``[basic.fundamental]``
section of the ISO C++ Standard for information about integral types.
```
consistently with other files in this patch

**Review by akukanov — APPROVED** [review]

**Pull request merged**